components/NewOfferForm/index.js

```diff
@@ -1,14 +1,26 @@
-export default function NewOfferForm({ credit, id }) {
+import useUser from "@/hooks/useUser";
+
+export default function NewOfferForm({ credit, id, onSubmit, user, starship }) {
+  function _onSubmit(event) {
```

Comment:
Convention in javascript is not to start functions with `u`.

An alternative name could be `handleSubmit`